src/Writer.php

```diff
@@ -68,18 +78,87 @@ public function testConnection(): void
 
     public function write(): void
     {
-        // TODO
+        // Create CSV reader
+        $this->logger->info(sprintf(
+            'Exporting table "%s" in "%s" mode ...',
+            $this->config->getTableId(),
+            $this->config->getMode()
+        ));
+
+        // Create mapper
+        $mapper = $this->messageMapperFactory->create();
+
+        // Register a new NodeJs process to event loop.
+        $process = $this->createNodeJsProcess('write.js');
+        $messageStream = $process->getMessageStream();
+        $messageWriter = new MessageWriter($messageStream);
+
+        // Throw an exception on process failure
+        $process
+            ->getPromise()
+            ->done(null, function (Throwable $e): void {
+                if ($e instanceof ProcessException && $e->getExitCode() === 1) {
+                    throw new UserException('Export failed.', $e->getCode(), $e);
+                } else {
+                    throw new ApplicationException($e->getMessage(), $e->getCode(), $e);
+                }
+            });
+
+        // Schedule the first batch
+        $this->futureWriteCsvRows($mapper->getMessages(), $messageWriter);
+
+        // Start event loop
+        $this->loop->run();
+
+        // Done
+        $this->logger->info(sprintf(
+            'Exported all %d rows from the table "%s".',
+            $messageWriter->getCount(),
+            $this->config->getTableId()
+        ));
+    }
+
+    protected function writeCsvRows(Iterator $messages, MessageWriter $messageWriter): void
+    {
+        if ($messageWriter->isBufferFull()) {
+            usleep(5000); // wait 5ms and check again
```

Comment:
Ak je buffer plny tak pockame 5ms, ... a cez futureTick sa to spusti opat ... inak by to neustale kontrolovalo a vytazilo CPU na 100%.